src/model/Timestep.cpp

```diff
@@ -66,11 +68,19 @@ namespace model {
     }
 
     bool Timestep_Impl::setNumberOfTimestepsPerHour(int numberOfTimestepsPerHour) {
+      if (numberOfTimestepsPerHour != this->numberOfTimestepsPerHour()) {
+        for (auto& sch_day : model().getConcreteModelObjects<ScheduleDay>()) {
+          sch_day.getImpl<ScheduleDay_Impl>()->clearCachedTimeSeries();
+        }
+      }
```

Comment:
Reset the cachedTImeSeries when we change the number of timesteps per hour in both Timestep::setNumberOfTimesPerHour and the resetNum... ok!